Authoring/WinRT.SourceGenerator/WinRTTypeWriter.cs

```diff
@@ -0,0 +1,1791 @@
+            : this(new Symbol(type), name, attributes)
+        {
+        }
+
+        public Parameter(ParameterSyntax parameter, ParameterAttributes attributes, SemanticModel model)
+        {
+            Type = new Symbol(model.GetDeclaredSymbol(parameter).Type);
+            Name = parameter.Identifier.ValueText;
+            Attributes = attributes;
+        }
+    }
+
+    class Symbol
+    {
+        public ITypeSymbol Type;
+        public EntityHandle Handle;
+
+        public Symbol(ITypeSymbol type)
+        {
+            Type = type;
+            Handle = default;
+        }
+
+        public Symbol(EntityHandle handle)
+        {
+            Type = default;
+            Handle = handle;
+        }
+
+        public bool IsHandle()
+        {
+            return Handle != default;
+        }
+    }
+
+    class TypeDeclaration
+    {
+        public readonly ISymbol Node;
+        public TypeDefinitionHandle Handle;
+        public string DefaultInterface;
+
+        public Dictionary<ISymbol, List<MethodDefinitionHandle>> MethodDefinitions;
+        public Dictionary<ISymbol, List<EntityHandle>> MethodReferences;
+        public Dictionary<ISymbol, FieldDefinitionHandle> FieldDefinitions;
+        public Dictionary<ISymbol, PropertyDefinitionHandle> PropertyDefinitions;
+        public Dictionary<ISymbol, EventDefinitionHandle> EventDefinitions;
+
+        public TypeDeclaration(ISymbol node)
+        {
+            Node = node;
+            MethodDefinitions = new Dictionary<ISymbol, List<MethodDefinitionHandle>>();
+            MethodReferences = new Dictionary<ISymbol, List<EntityHandle>>();
+            FieldDefinitions = new Dictionary<ISymbol, FieldDefinitionHandle>();
+            PropertyDefinitions = new Dictionary<ISymbol, PropertyDefinitionHandle>();
+            EventDefinitions = new Dictionary<ISymbol, EventDefinitionHandle>();
+        }
+
+        public override string ToString()
+        {
+            return Node.ToString();
+        }
+
+        public void AddMethod(ISymbol node, MethodDefinitionHandle handle)
+        {
+            if(!MethodDefinitions.ContainsKey(node))
+            {
+                MethodDefinitions[node] = new List<MethodDefinitionHandle>();
+                MethodReferences[node] = new List<EntityHandle>();
+            }
+
+            MethodDefinitions[node].Add(handle);
+            MethodReferences[node].Add(handle);
+        }
+
+        public void AddMethodReference(ISymbol node, MemberReferenceHandle handle)
+        {
+            if (!MethodReferences.ContainsKey(node))
+            {
+                MethodReferences[node] = new List<EntityHandle>();
+            }
+
+            MethodReferences[node].Add(handle);
+        }
+
+        public List<MethodDefinitionHandle> GetMethodDefinitions()
+        {
+            return MethodDefinitions.Values.SelectMany(list => list).ToList();

```

Comment:
can't just use MethodDefinitions.Values.ToList() ?

Reply:
No, that makes it a list of lists.  I am trying to merge all the method definitions created for each symbol (i.e. property, method, event, etc.) into 1 list.